airflow/kubernetes/pod_generator.py

```diff
@@ -430,6 +419,70 @@ def construct_pod(
         except Exception as e:
             raise PodReconciliationError from e
 
+    @classmethod
+    def build_selector_for_k8s_executor_pod(
+        cls,
+        *,
+        dag_id,
+        task_id,
+        try_number,
+        map_index=None,
+        date=None,
+        run_id=None,
+    ):
+        """
+        Generate selector for kubernetes executor pod
+
+        :meta private:
+        """
+        labels = cls.build_labels_for_k8s_executor_pod(
+            dag_id=dag_id,
+            task_id=task_id,
+            try_number=try_number,
+            map_index=map_index,
+            date=date,
+            run_id=run_id,
+        )
+        label_strings = [f"{label_id}={label}" for label_id, label in sorted(labels.items())]
+        selector = ",".join(label_strings)
+        selector += ",airflow-worker"
+        return selector
+
+    @classmethod
+    def build_labels_for_k8s_executor_pod(
+        cls,
+        *,
+        dag_id,
+        task_id,
+        try_number,
+        airflow_worker=None,
+        map_index=None,
+        date=None,
+        run_id=None,
+    ):
+        """
+        Generate labels for kubernetes executor pod
+
+        :meta private:
+        """
+        labels = {
+            "dag_id": make_safe_label_value(dag_id),
+            "task_id": make_safe_label_value(task_id),
+            "try_number": str(try_number),
+            "kubernetes_executor": "True",
+        }
+        if airflow_worker is not None:
```

Comment:
I wonder if we should have `build_selector_for_k8s_executor_pod` built it's own dict so we don't have to do all these conditional checks - it took me a second to figure out why it was this way. Plus it'll ensure we set everything we actually want to on the pod (e.g. airflow_version).

Reply:
confused a bit...
but...
the reason i want to reuse a shared labels func is... the way we generate selector must agree with the way we generate the labels -- if there is inconsistency then we may not find the pod. only way to guarantee consistent is to use same func

Reply:
Right, good point. Let's pull the heavy lifting with these optional args into a private function then, and have the public `build_labels_for_k8s_executor_pod` require them? That should cover everything?

Reply:
i have both of these methods set to private...  (TP style, `:meta private:`) and .... `build_labels_for_k8s_executor_pod` _is_ the "heavy lifting" one IMO... clarify?

Reply:
we do basically same thing for KPO...

Reply:
I should have air quoted, "public". Where I'm going is that it feels weird to me having `build` even work without all of the labels we'd set on the pod being required, moving the "what's really required" outside of PodGenerator. I'm coming at it from the PodGenerator "public internal" interface perspective.

e.g.
```
def _get_labels_dict(
    *,
    dag_id,
    task_id,
    try_number,
    airflow_worker=None,
    map_index=None,
    date=None,
    run_id=None,
):
    ...

def build(
    *,
    dag_id,
    task_id,
    try_number,
    airflow_worker,
    map_index,
    date,
    run_id,
):
    return _get_labels_dict(...)

def selector(*, dag_id, task_id, try_number):
   return _get_labels_dict(...)
```

This seems to more easily mesh with what you should to do for build vs get the selector (e.g. code completion/typing - ":meta private:" is for sphinx after all).

However, enough bikeshedding on my part :)

Reply:
I don't mind the discussion

I *think* I understand where you're coming from

The way I see it, in this case, the caller must know what it needs and it's not the labels function to decide. It just builds requested labels in the right way.  If we make the function decide what is required, that's sort of a "logic in two places" kind of thing. And so, that's why I'm able to use the one labels func in two ways. It's not worried about the requirements -- that's the callers job, and different callers have different needs -- one is for pod creation and the other for pod finding.

Re private, I think it's @uranusjr 's position that meta private means not part of the public API, not just in terms of doc but also backcompat, and so that's what I went with here.  And it has the pleasant side effect of no warning about private method accessed.

Reply:
But, i should add... honestly don't feel strongly.  And particularly since the intention is to make these methods non-public (i.e. internal, no backcompat, therefore no friction to change in future) then happy to code it up however.

Reply:
Just different design decisions ultimately if the caller must know what's require or the callable is able to enforce what's required. Not a big deal.